clar2wasm/src/deserialize.rs

```diff
@@ -962,220 +963,349 @@ impl WasmGenerator {
-                        then.br(block_id);
+                    MemArg {
+                        align: 1,
+                        offset: 0,
                     },
-                    |_| {},
                 );
 
-            // Compare the key to the expected key
-            let key_bytes = key.as_bytes();
-            for (i, byte) in key_bytes.iter().enumerate() {
-                block
-                    .local_get(offset_local)
-                    .load(
-                        memory,
-                        LoadKind::I32_8 {
-                            kind: ExtendedLoad::ZeroExtend,
-                        },
-                        MemArg {
-                            align: 1,
-                            offset: i as u32 + 1,
-                        },
-                    )
-                    .i32_const(*byte as i32)
+                // Verify the prefix byte
+                done_block
+                    .i32_const(TypePrefix::Tuple as i32)
                     .binop(BinaryOp::I32Ne)
-                    .if_else(
-                        None,
-                        |then| {
-                            // Return none
-                            then.i32_const(0);
-                            add_placeholder_for_clarity_type(then, &ty);
-                            then.br(block_id);
-                        },
-                        |_| {},
-                    );
-            }
+                    .br_if(done_block_id);
 
-            // Increment the offset by the key length and its size
-            block
-                .local_get(offset_local)
-                .i32_const(key.len() as i32 + 1)
-                .binop(BinaryOp::I32Add)
-                .local_set(offset_local);
-
-            // Deserialize the value. Note, this will update the offset to
-            // point to the next key.
-            self.deserialize_from_memory(&mut block, offset_local, end_local, value_ty)?;
+                // Read the number of keys and check that it's >= to the
+                // result tuple number of fields
+                done_block
+                    .local_get(offset_local)
+                    .i32_const(1)
+                    .binop(BinaryOp::I32Add)
+                    .call(self.func_by_name("stdlib.load-i32-be"))
+                    .local_tee(remaining_fields)
+                    .i32_const(tuple_ty.get_type_map().len() as i32)
+                    .binop(BinaryOp::I32LtU)
+                    .br_if(done_block_id);
 
-            // Check if the deserialization failed:
-            // - Store the value in locals
-            // - Check the inidicator now on top of the stack
-            let inner_locals = self.save_to_locals(&mut block, value_ty, true);
+                // Update the offset to point to the first key
+                done_block
+                    .local_get(offset_local)
+                    .i32_const(5)
```

Comment:
Why an integer `5` is being added to the `offset_local`? Can you please clarify that with a comment?

Reply:
Because I update the offset to point to the first key? What would you add?